File: Easy/350-Intersection of Two Arrays II/350.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """

        dict1 = {}
        for n in nums1:
            dict1[n] = dict1.get(n, 0) + 1
        
        intersection = []
        for n in nums2:
            if n in dict1 and dict1[n] > 0:
                intersection.append(n)
                dict1[n] -= 1
        
        return intersection

File: Easy/350-Intersection of Two Arrays II/350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        
        unordered_map<int, int> map1;
        for(int i = 0; i < nums1.size(); ++i)
            map1[nums1[i]]++;
        
        vector<int> intersection;
        for(int i = 0; i < nums2.size(); ++i)
            if(--map1[nums2[i]] >= 0)
                intersection.push_back(nums2[i]);
            
        return intersection;
    }
};